source/sdk/threading/Thread.ooc

```diff
@@ -165,6 +172,26 @@ Mutex: abstract class {
 
 }
 
+MutexUnsafe: class extends Mutex {
+	init: func
+	lock: override func
+	unlock: override func
+}
+
+MutexGlobal: class extends Mutex {
+	_globalMutex := static Mutex new(MutexType Safe)
+	init: func
+	lock: override func {
+		This _globalMutex lock()
+	}
+	unlock: override func {
+		This _globalMutex unlock()
+	}
+	cleanup: static func {
```

Comment:
I find "I suggest that we turn Closure from a cover (struct) into a class." in a very big issue from June 2012: https://github.com/fasterthanlime/rock/issues/439